src/foundry/foundry.js/clientDocuments/actor.d.ts

```diff
@@ -25,10 +26,258 @@ declare global {
+
+    /**
+     * An object that tracks which tracks the changes to the data model which were applied by active effects
+     * @defaultValue `{}`
+     */
+    overrides: Record<string, unknown>;
+
+    /**
+     * A cached array of image paths which can be used for this Actor's token.
+     * Null if the list has not yet been populated.
+     * @defaultValue `null`
+     */
+    protected _tokenImages: string[] | null;
+
+    /**
+     * Cache the last drawn wildcard token to avoid repeat draws
+     * @defaultValue `null`
+     */
+    protected _lastWildcard: string | null;
+
     /**
      * A convenient reference to the file path of the Actor's profile image
      */
     get img(): this['data']['img'];
+
+    /**
+     * Provide an object which organizes all embedded Item instances by their type
+     */
+    get itemTypes(): Record<string, InstanceType<ConfiguredDocumentClass<typeof foundry.documents.BaseItem>>>;
+
+    /**
+     * Test whether an Actor entity is a synthetic representation of a Token (if true) or a full Entity (if false)
+     */
+    get isToken(): boolean;
+
+    /**
+     * An array of ActiveEffect instances which are present on the Actor which have a limited duration.
+     */
+    get temporaryEffects(): InstanceType<ConfiguredDocumentClass<typeof foundry.documents.BaseActiveEffect>>[];
+
+    /**
+     * Return a reference to the TokenDocument which owns this Actor as a synthetic override
+     */
+    get token(): InstanceType<ConfiguredDocumentClass<typeof foundry.documents.BaseToken>> | null;
+
+    /**
+     * A convenience reference to the item type (data.type) of this Actor
+     */
+    get type(): this['data']['type'];
+
+    /** @override */
+    get uuid(): string;
+
+    /**
+     * Apply any transformations to the Actor data which are caused by ActiveEffects.
+     */
+    applyActiveEffects(): void;
+
+    /**
+     * Retrieve an Array of active tokens which represent this Actor in the current canvas Scene.
+     * If the canvas is not currently active, or there are no linked actors, the returned Array will be empty.
+     * If the Actor is a synthetic token actor, only the exact Token which it represents will be returned.
+     *
+     * @param linked   - Limit results to Tokens which are linked to the Actor. Otherwise return all Tokens even those which are not linked. (default: `false`)
+     * @param document - Return the Document instance rather than the PlaceableObject (default: `false`)
+     * @returns An array of Token instances in the current Scene which reference this Actor.
+     */
+    getActiveTokens(
+      linked?: boolean,
+      document?: boolean
+    ): InstanceType<ConfiguredDocumentClass<typeof foundry.documents.BaseToken>>[];
+
+    /**
+     * Prepare a data object which defines the data schema used by dice roll commands against this Actor
+     */
+    getRollData(): this['data']['data'];
+
+    /** @override */
+    protected _getSheetClass(): ConstructorOf<FormApplication>;
```

Comment:
I think we can reference `ActorSheet` here? Or this this because `ActorSheet` has not yet been updated? If so, please add a TODO.

For `Item`, I typed this as
```ts
     protected _getSheetClass(): typeof ItemSheet;
```
But I'm not 100% sure if `ConstructorOf` is needed instead.

Reply:
Largely it was me just forwarding ClientDocumentMixin's def (I overlooked the null) and Foundry's code kinda does not care that it's anything more than a FormApplication. Alternate sheet renderers might have a differing inheritance tree, either doing a mixin style approach (which should be types compatible), or opting to have it's own base class extending FormApplication and it's own set of replacement ItemSheet and ActorSheet. (Vue and Svelte come to mind as renders to replace the default system as they have seen some use in Foundry already, though I don't know how they are done off the top of my head.)

Reply:
Shouldn't this be typed to the configured sheet class?